kroxylicious-integration-tests/src/test/java/io/kroxylicious/kms/service/KmsIT.java

```diff
@@ -0,0 +1,164 @@
+        try {
+            var service = (KmsService<C, K, E>) facade.getKmsServiceClass().getDeclaredConstructor(new Class[]{}).newInstance();
+            kms = service.buildKms((C) facade.getKmsServiceConfig());
+        }
+        catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+        manager = facade.getTestKekManager();
+    }
+
+    @TestTemplate
+    void resolveKeyByName() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .isEqualTo(alias);
+    }
+
+    @TestTemplate
+    void resolveWithUnknownKey() {
+        var alias = "unknown";
+        var resolved = kms.resolveAlias(alias);
+        assertThat(resolved)
+                .failsWithin(Duration.ofSeconds(5))
+                .withThrowableThat()
+                .withCauseInstanceOf(UnknownAliasException.class);
+    }
+
+    @TestTemplate
+    void generatedEncryptedDekDecryptsBackToPlain() {
+        var alias = "mykey";
+        manager.generateKek(alias);
+
+        var resolveStage = kms.resolveAlias(alias);
+        assertThat(resolveStage).succeedsWithin(Duration.ofSeconds(5));
+
+        var pairStage = kms.generateDekPair(resolveStage.toCompletableFuture().join());
+        assertThat(pairStage).succeedsWithin(Duration.ofSeconds(5));
+        var pair = pairStage.toCompletableFuture().join();
+
+        var decryptedDekStage = kms.decryptEdek(pair.edek());
+        assertThat(decryptedDekStage)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .matches(sk -> SecretKeyUtils.same((DestroyableRawSecretKey) sk, (DestroyableRawSecretKey) pair.dek()));
+    }
+
+    @TestTemplate
+    void generatedDekPairWithUnknownKey() {
+        var alias = "mykey";
+        var key = createAndResolve(alias);
+
+        manager.deleteKek(alias);
+
+        var pairStage = kms.generateDekPair(key);
+        assertThat(pairStage)
+                .failsWithin(Duration.ofSeconds(5))
+                .withThrowableThat()
+                .withCauseInstanceOf(UnknownKeyException.class);
+    }
+
+    @TestTemplate
+    void decryptDekAfterRotate() {
+        var alias = "mykey";
+        var key = createAndResolve(alias);
+
+        var pairStage = kms.generateDekPair(key);
+        assertThat(pairStage).succeedsWithin(Duration.ofSeconds(5));
+        var pair = pairStage.toCompletableFuture().join();
+
+        manager.rotateKek(alias);
+
+        var decryptedDekStage = kms.decryptEdek(pair.edek());
+        assertThat(decryptedDekStage)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .matches(sk -> SecretKeyUtils.same((DestroyableRawSecretKey) sk, (DestroyableRawSecretKey) pair.dek()));
+    }
+
+    @TestTemplate
+    void decryptEdekWithUnknownKey() {
```

Comment:
The name suggests to me it should succeed when failure is actually expected. 
```suggestion
    void failToDecryptEdekWithUnknownKey() {
```